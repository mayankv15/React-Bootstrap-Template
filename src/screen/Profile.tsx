import React, { useState } from "react";
import { Typography } from "@mui/material";
import keycloak from "../keycloak";
import Button from "../components/common/controls/Button";
import Input from "../components/common/controls/Input";
import SIZE from "../theme/colors.json";
import COLORS from "../theme/colors.json";
import { request } from "../services/request";
import CONSTANTS from "../constants/constants";
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles(() => ({
  label: {
    marginBottom: SIZE[0],
    fontSize: SIZE[15],
    textAlign: "left",
    color: COLORS.GRAY,
  },
}));

const Profile = () => {
  const classes = useStyles();
  const [updateEnable, setUpdateEnbale] = useState(true);
  const size = "small";
  const variant = "standard";
  const [firstName, setFirstName] = useState(
    sessionStorage.getItem(CONSTANTS.FIRST_NAME)
  );
  const [lastName, setLastName] = useState(
    sessionStorage.getItem(CONSTANTS.LAST_NAME)
  );
  const UPDATE_PROFILE_URL = `auth/admin/realms/techsophy-platform/users/`;
  const USER_ID = `${sessionStorage.getItem(CONSTANTS.USER_ID)}`;
  const USER_MAIL = `${sessionStorage.getItem(CONSTANTS.USER_EMAIL)}`;
  const { idTokenParsed } = keycloak;

  let formBody = {
    id: USER_ID,
    username: USER_MAIL,
    enabled: true,
    firstName: firstName,
    lastName: lastName,
    requiredActions: ["UPDATE_PROFILE"],
    access: {
      manageGroupMembership: true,
      view: true,
      mapRoles: true,
      impersonate: true,
      manage: true,
    },
  };

  const handleProfileUpdate = async () => {
    try {
      await request
        .put(
          `${process.env.REACT_APP_KEYCLOAK_URL}` +
            UPDATE_PROFILE_URL +
            USER_ID,
          formBody
        )
        .then(() => {
          sessionStorage.setItem(
            CONSTANTS.FIRST_NAME,
            idTokenParsed.given_name
          );
          sessionStorage.setItem(
            CONSTANTS.LAST_NAME,
            idTokenParsed.family_name
          );
          setUpdateEnbale(true);

          alert("User has been updated successfully!");
        });
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <div
      style={{ display: "flex", justifyContent: "center", paddingTop: "50px" }}
    >
      <Typography variant="h3">
        <Input
          label="User Email"
          value={sessionStorage.getItem("user-email")}
          name="userEmail"
          type="text"
          size={size}
          inputLabelStyles={classes.label}
          disabled
          required
          variant={variant}
        />
        <Input
          label="First Name"
          value={firstName}
          name="first-name"
          type="text"
          size={size}
          inputLabelStyles={classes.label}
          required
          variant={variant}
          onChange={(e) => {
            setUpdateEnbale(false);
            setFirstName(e.target.value);
          }}
        />
        <Input
          label="Last Name"
          value={lastName}
          name="last-name"
          type="text"
          size={size}
          inputLabelStyles={classes.label}
          required
          variant={variant}
          onChange={(e) => {
            setUpdateEnbale(false);
            setLastName(e.target.value);
          }}
        />
        <div
          style={{
            display: "flex",
            justifyContent: "center",
            padding: 20,
            margin: 20,
          }}
        >
          <Button
            text="Update"
            size={size}
            disabled={updateEnable}
            onClick={() => {
              handleProfileUpdate();
            }}
          />
        </div>
      </Typography>
    </div>
  );
};

export default Profile;
