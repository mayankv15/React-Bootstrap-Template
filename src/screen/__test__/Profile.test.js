import React from "react";
import { render } from "@testing-library/react";
import Profile from "../Profile"

describe("Renders Profile", () => {
    it("should render Profile correctly", () => {
        const component = render(<Profile />);
        expect(component).toMatchSnapshot();
    });
});
