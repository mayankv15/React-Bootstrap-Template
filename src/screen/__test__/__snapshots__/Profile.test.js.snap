// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Profile should render Profile correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; justify-content: center; padding-top: 50px;"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          <div
            class="form-control"
          >
            <p
              class="MuiTypography-root makeStyles-label-1 undefined MuiTypography-body1"
            >
              User Email
               
              *
            </p>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  disabled=""
                  name="userEmail"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="form-control"
          >
            <p
              class="MuiTypography-root makeStyles-label-1 undefined MuiTypography-body1"
            >
              First Name
               
              *
            </p>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  name="first-name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="form-control"
          >
            <p
              class="MuiTypography-root makeStyles-label-1 undefined MuiTypography-body1"
            >
              Last Name
               
              *
            </p>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  name="last-name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            style="display: flex; justify-content: center; padding: 20px; margin: 20px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root makeStyles-root-5 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label makeStyles-label-6"
              >
                
                Update
              </span>
            </button>
          </div>
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; justify-content: center; padding-top: 50px;"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        <div
          class="form-control"
        >
          <p
            class="MuiTypography-root makeStyles-label-1 undefined MuiTypography-body1"
          >
            User Email
             
            *
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                disabled=""
                name="userEmail"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="form-control"
        >
          <p
            class="MuiTypography-root makeStyles-label-1 undefined MuiTypography-body1"
          >
            First Name
             
            *
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                name="first-name"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="form-control"
        >
          <p
            class="MuiTypography-root makeStyles-label-1 undefined MuiTypography-body1"
          >
            Last Name
             
            *
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                name="last-name"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          style="display: flex; justify-content: center; padding: 20px; margin: 20px;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-5 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-label-6"
            >
              
              Update
            </span>
          </button>
        </div>
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
